Initial getting started guide

dack-fe opened the pull request:
This is an initial getting started guide for installing and configure the Autogen Software Flow.  The linux-socfpga installation and configuration steps are done in the WSL or Linux home directory. This affects Pull Request #48 in the Simulink Code Gen Repo. (https://github.com/fpga-open-speech-tools/simulink_codegen/pull/48)

dack-fe committed:
Initial Commit

dack-fe committed:
Initial Commit of Clone Bash Script

dack-fe committed:
Updated the Installation Step

dack-fe committed:
Initial Commit

dack-fe committed:
Updated

dack-fe committed:
Added AWS Configure

dack-fe committed:
Added the steps to generate an example design

tvannoy commented on dtc-overlay.sh:
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh -e
```
So this script just installs `dtc`? If so, let's rename it to `dtc-install.sh` or something similar.

dack-fe replied:
Updated to dtc_installer.sh

tvannoy commented on getting_started.md:
```diff
@@ -0,0 +1,62 @@
+## Clone the Git Repos
+ 1. Clone the Docs Repos
+ 2. Run `./docs/cloneNIH.sh` from WSL
+ 3. Run `./docs/dtc-overlay.sh` from WSL - This will fail to build a Python 2 package but that doesn't matter
```
If `dtc-overlay.sh` installs `dtc`, then `sudo apt install device-tree-compiler -y` is not needed.

dack-fe replied:
Removed in new FrOST_Autogen_Config.sh

tvannoy commented on getting_started.md:
```diff
@@ -0,0 +1,62 @@
+#### Setting Up the Linux SoC-FPGA on WSL
+1. `cd ~; git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git`
+2. `cd linux-socfpga`
+3. `cp arch/arm/configs/openspeech_defconfig .config`
+4. `make menu config ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-` - Save and Exit
```
This functionality can be replaced with `make openspeech_defconfig`

dack-fe replied:
I attempted to replace the two lines above with `make openspeech_defconfig;` in `FrOST_Autogen_Config.sh` but got the following error
![linux-socfpga-error](https://user-images.githubusercontent.com/53094583/94457846-5e01ae00-0172-11eb-83fe-3967107aeb5b.PNG)

dack-fe replied:
Reverted to copying the config and running make menuconfig - Commit 4f63467

tvannoy replied:
Oops, we need to export the cross compile environment variables. So either 
```bash
export ARCH=arm
export CROSS_COMPILE=arm-linux-gnueabihf
make openspeech_defconfig
```
or
```bash
make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf openspeech_defconfig
```

Both of those should hopefully work. In your screenshot, it was trying to pull the config file from arch/x86/configs, but it exists in arch/arm/configs.

dack-fe replied:
Updated with Commit 29f0c4c

tvannoy commented on getting_started.md:
```diff
@@ -0,0 +1,62 @@
+2. Simulink                  - Required  
+3. HDL Coder                 - Required  
+4. DSP System Toolbox        - Strongly Suggested, Required for Example Designs  
+5. Fixed-Point Designer      - Strongly Suggested, Required for Example Designs  
```
I wonder if Fixed-Point Designer is actually required. I guess if people could make designs without fixed-point objects in Simulink, it's not strictly needed, though in practice it probably is.

dack-fe replied:
Updated MATLAB Requirements - https://www.mathworks.com/support/requirements/hdl-coder.html
Commit: e234329

dack-fe committed:
Renamed the Clone Repos Files

dack-fe committed:
Renamed the DTC Install Script

dack-fe committed:
Updated with steps to use the FrOST Autogen Config Script

dack-fe committed:
Initial Commit

Testing the Linux-SOCFPGA Configure Steps

dack-fe committed:
Added the Save and Exit step during make menuconfig

dack-fe committed:
Added WSL Configure and Repo Clone back in

dack-fe committed:
Updated the requirements for MATLAB Toolboxes

dack-fe committed:
Moved to scripts folder

dack-fe committed:
Updated with the make openspeech_defconfig

dack-fe committed:
Removed libncurses, added print comments

Removed Libncurses-dev: This was originally need for the make menuconfig but was replaces with the openspeech_defconfig step
Added additional echo statements to inform the user of what is going on

fe-tdavis committed:
Removed /bin/sh line and return characters

tvannoy commented:
@fe-tdavis Why remove the `#!/bin/sh` lines? Those are used so the scripts can be run as `./script_name.sh` instead of `bash script_name.sh`. If anything, we should probably set those lines to `#!/bin/bash` just incase we are using any bash-specific features.

dack-fe committed:
Added #!/bin/bash

fe-wickham commented on FrOST_Autogen_Config.sh:
```diff
@@ -0,0 +1,25 @@
+
+# Clone and Configure Linux-SOCFPGA
+cd ~; echo "Moving to Home Directory of WSL."
+#git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git;
```
Uncomment and check for the repo already existing
```suggestion
[ ! -d 'linux-socfpga' ] && git clone https://github.com/fpga-open-speech-tools/linux-socfpga.git;
```

dack-fe replied:
Sounds good. @fe-tdavis had the linux repo clone commented out for testing.

fe-wickham commented on scripts/dtc_installer.sh:
```diff
@@ -0,0 +1,60 @@
+	fi
+
+	make clean
+	make PREFIX=/usr/local/ CC=gcc CROSS_COMPILE= all
```
```suggestion
	make PREFIX=/usr/local/ CC=gcc CROSS_COMPILE= all NO_PYTHON=1
```

This makes it so it won't attempt to build the Python library that fails

fe-wickham reviewed (changes requested):
Looks good overall but cloning the linux repo needs to be uncommented

dack-fe committed:
Uncomment and check for the repo already existing

Co-authored-by: Dylan Wickham <61513109+fe-wickham@users.noreply.github.com>

dack-fe committed:
Added flag to prevent python 2 compile that fails

Co-authored-by: Dylan Wickham <61513109+fe-wickham@users.noreply.github.com>

fe-wickham reviewed (approved):
Looks good now

tvannoy reviewed (approved):
Looks fine to me as well. This is a good start, for sure.

dack-fe committed:
Checking if the repo folder exisit before attempting to clone

This is a simple check for the repo folder. It doesn't look at what is inside - ie the correct files or tags.

fe-tdavis commented:
> @fe-tdavis Why remove the `#!/bin/sh` lines? Those are used so the scripts can be run as `./script_name.sh` instead of `bash script_name.sh`. If anything, we should probably set those lines to `#!/bin/bash` just incase we are using any bash-specific features.

@tvannoy The change was made because when the script was run on a fresh install of Ubuntu 20.04 on the WSL, an error is produced.  In fact, the script could still be run on the problematic machine using `./`, however this is likely due to the default interpreter being set to bash.

tvannoy commented:
> @tvannoy The change was made because when the script was run on a fresh install of Ubuntu 20.04 on the WSL, an error is produced. In fact, the script could still be run on the problematic machine as `./`, however this is likely due to the default interpreter being set to bash.

@fe-tdavis Gotcha. Do you remember what the error was, by chance? Did changing it to `#!/bin/bash` fix the error? I believe dash is the [default /bin/sh in Ubuntu](https://wiki.ubuntu.com/DashAsBinSh) still, though bash is still the default login shell.

dack-fe committed:
Fixed dos character errors, updated check for website repo

dack-fe committed:
Update to bin/sh

dack-fe committed:
Updated EOL Characters, bin/bash

dack-fe committed:
Updated the getting started guide

dack-fe committed:
Added sudo for WSL2

dack-fe committed:
Added dos2unix command

dack-fe committed:
Added steps to ensure the script files have DOS EOL characters

dack-fe committed:
Removed duplicate

dack-fe committed:
Added notes about Python 3

dack-fe commented:
Any final comments? @fe-tdavis @tvannoy @fe-wickham
